main.tf

```
@@ -7,11 +18,35 @@ terraform {
   }
 }
 
+
 resource "random_string" "deployment_id" {
-  count = "${var.deployment_id != "" ? 0 : 1}"
+  count = local.create_deployment_id
 
   length   = 8
   special  = false
   upper = false
   numeric = false
+
+}
+
+#
+# AWS availability zones
+#
+data "aws_availability_zones" "available_azs" {
+  state = "available"
+  filter {
+    name = "opt-in-status"
+    # Currently, no support for Local Zones, Wavelength, or Outpost
+    values = ["opt-in-not-required"]
+  }
+}
+#
+# VPC
+#
+module "vpc" {
+  depends_on = [random_string.deployment_id]
+
+  source = "./modules/vpc"
+  deployment_id = var.deployment_id != "" ? var.deployment_id : random_string.deployment_id[0].id
```

```suggestion
  deployment_id = var.deployment_id == "" ? random_string.deployment_id[0].id : var.deployment_id
```
This matches the previous logic in output.tf and is not a negative conditional by default.